src/common/src/array/arrow/arrow_iceberg.rs

```diff
@@ -97,6 +97,7 @@ impl ToArrow for IcebergArrowConvert {
             DataType::Date => self.date_type_to_arrow(),
             DataType::Time => self.time_type_to_arrow(),
             DataType::Timestamp => self.timestamp_type_to_arrow(),
+            DataType::TimestampNanosecond => self.timestampns_type_to_arrow(),
```

Comment:
Here we only handle sink timestamp_ns to iceberg, however, the opposite direction, read timestamp_ns from iceberg should be handled as well. I think we can add this new data type to the iceberg engine full-type test. https://github.com/risingwavelabs/risingwave/blob/ae86ddccbdd740386038430ba6eb36ca1ff29bc5/e2e_test/iceberg/test_case/iceberg_engine.slt#L27